rows.go

```diff
@@ -31,49 +31,71 @@ type Row struct {
 
 // Cell represents the data in a single cell as a consumable format.
 type Cell struct {
-	Column string // E.G   A, B, C
-	Row    int
-	Value  string
+	Column      string // E.G   A, B, C
+	ColumnIndex int
+	Row         int
+	Value       string
+	Type        CellType
 }
 
+type CellType int
+
+const (
+	TypeNil CellType = iota
+	TypeString
+	TypeNumerical
+	TypeDateTime
+	TypeDate
+	TypeBoolean
+)
+
 // getCellValue interrogates a raw cell to get a textual representation of the cell's contents.
 // Numerical values are returned in their string format.
 // Dates are returned as an ISO YYYY-MM-DD formatted string.
-// Datetimes are returned in RFC3339 (ISO-8601) YYYY-MM-DDTHH:MM:SSZ formated string.
-func (x *XlsxFile) getCellValue(r rawCell) (string, error) {
+// Datetimes are returned in RFC3339 (ISO-8601) YYYY-MM-DDTHH:MM:SSZ formatted string.
+func (x *XlsxFile) getCellValue(r rawCell) (string, CellType, error) {
 	if r.Type == "inlineStr" {
 		if r.InlineString == nil {
-			return "", fmt.Errorf("Cell had type of InlineString, but the InlineString attribute was missing")
+			return "", TypeNil, fmt.Errorf("Cell had type of InlineString, but the InlineString attribute was missing")
 		}
-		return *r.InlineString, nil
+		return *r.InlineString, TypeString, nil
 	}
 
 	if r.Value == nil {
-		return "", fmt.Errorf("Unable to get cell value for cell %s - no value element found", r.Reference)
+		return "", TypeNil, fmt.Errorf("Unable to get cell value for cell %s - no value element found", r.Reference)
 	}
 
 	if r.Type == "s" {
 		index, err := strconv.Atoi(*r.Value)
 		if err != nil {
-			return "", err
+			return "", TypeNil, err
 		}
 		if len(x.sharedStrings) <= index {
-			return "", fmt.Errorf("Attempted to index value %d in shared strings of length %d",
+			return "", TypeNil, fmt.Errorf("Attempted to index value %d in shared strings of length %d",
 				index, len(x.sharedStrings))
 		}
 
-		return x.sharedStrings[index], nil
+		return x.sharedStrings[index], TypeString, nil
 	}
 
 	if x.dateStyles[r.Style] && r.Type != "d" {
-		formattedDate, err := convertExcelDateToDateString(*r.Value)
+		formattedDate, typ, err := convertExcelDateToDateString(*r.Value)
```

Comment:
This might read better if the type was derived in a different function - `convertDatetoDateString` doesn't suggest that it's also parsing the type. Might need a bit of a refactor around here.